[Windows] Allow bmalloc build for Windows platform.

basuke opened the pull request:
#### 7f2402e915287f01288f7061e71be2de40cc98a8
<pre>
[Windows] Allow bmalloc build for Windows platform.
<a href="https://bugs.webkit.org/show_bug.cgi?id=255943">https://bugs.webkit.org/show_bug.cgi?id=255943</a>

Reviewed by Yusuke Suzuki.

This is the patch for Windows bmalloc implementation part 1 of N.

On Windows port, USE_SYSTEM_MALLOC is hard coded and cannot enable by
cmake option USE_SYSTEM_MALLOC=Off. This patch removes the restriction
by removing that code.

Note the default value of USE_SYSTEM_MALLOC on Windows port is still on.
There&apos;s no actual effect unless passing the cmake arg explicitly.

* Source/WTF/wtf/PlatformUse.h:
* Source/bmalloc/CMakeLists.txt:
* Source/cmake/OptionsWin.cmake:

Canonical link: <a href="https://commits.webkit.org/263399@main">https://commits.webkit.org/263399@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/beaa446311fd1df22e59c5c85c861ba85877059f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4622 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4766 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5987 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/4675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4752 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4578 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/4913 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4568 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4679 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4040 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5992 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/2175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4030 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/8883 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/3728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4024 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4102 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5624 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/4264 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/4486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/3660 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4596 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4017 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/4029 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1145 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1101 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/8061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/4705 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4379 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1251 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/WTF/wtf/FastMalloc.cpp:
```diff
@@ -34,7 +34,8 @@
 
 #if OS(WINDOWS)
-#include <windows.h>
+#include <Windows.h>
```
Is `Windows.h` correct? I was thinking that it is `windows.h`.

basuke replied:
Really? I keep believing it is `Windows.h` for decades. Let me check.

basuke replied:
Actually this part cannot be compiled yet. I remove this part from the change and keep researching it.

basuke replied:
In WebKit code base, lowercase (260) vs Capitalized (5). Lowercase is more consistent in the codebase.

Constellation reviewed (approved):
r=me

basuke committed:
[Windows] Allow bmalloc build for Windows platform.
https://bugs.webkit.org/show_bug.cgi?id=255943

Reviewed by Yusuke Suzuki.

This is the patch for Windows bmalloc implementation part 1 of N.

On Windows port, USE_SYSTEM_MALLOC is hard coded and cannot enable by
cmake option USE_SYSTEM_MALLOC=Off. This patch removes the restriction
by removing that code.

Note the default value of USE_SYSTEM_MALLOC on Windows port is still on.
There's no actual effect unless passing the cmake arg explicitly.

* Source/WTF/wtf/PlatformUse.h:
* Source/bmalloc/CMakeLists.txt:
* Source/cmake/OptionsWin.cmake:

Canonical link: https://commits.webkit.org/263399@main
